src/snowflake/snowpark/column.py

```diff
@@ -226,16 +227,22 @@ class Column:
     """
 
     def __init__(
-        self, expr1: Union[str, Expression], expr2: Optional[str] = None
+        self, expr1: Union[str, Expression], expr2: Optional[str] = None, ast: Optional[proto.SpColumnExpr] = None
     ) -> None:
+        self._ast = proto.SpColumnExpr() if ast is None else ast
```

Comment:
This logic seems wrong. You should defer the construction of the AST to the specific cases below.